src/audio/pipeline/pipeline-xrun.c

```diff
@@ -53,6 +53,7 @@ static int pipeline_comp_xrun(struct comp_dev *current,
 /* recover the pipeline from a XRUN condition */
 int pipeline_xrun_recover(struct pipeline *p)
 {
+	pipeline_reset(p, p->source_comp);
```

Comment:
Thanks for the good explanation in commit message. So was this broken when we moved to delayed trigger processing?

Reply:
@kv2019i I don't think this would behave differently before the delayed trigger processing changes. As for this change itself - AFAIK we don't think we test xruns in CI, so this won't affect our CI results. And in fact it would be good to have this tested with both IPC3 and IPC4. But yes, I can confirm that often (or always) when xruns currently hit, the system is left in a very bad state, so if this improves that - it's good.